File: Contact.Application/DTOs/PhoneBookEmploymentDto.cs
using Contact.Domain.Enums;
using Core.Shared.Enums.Contact;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Contact.Application.DTOs
{
    
    public class ContactDetailDto
    {
        public string Title { get; set; } = null!;
        public string Value { get; set; } = null!;
        public ContactTypeEnum Type { get; set; }
        public ContactProfileTypeEnum Source { get; set; }
        
    }

    public class PhoneBookEmploymentDto
    {
        public string EmploymentCode { get; set; } = null!;
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string FullName => $"{FirstName} {LastName}".Trim();

        // اطلاعات سازمانی و شغلی
        public string? OrganizationUnitsName { get; set; }
        public string? HeadOfOrganizationUnitsName { get; set; }
        public string? JobTitleName { get; set; }
        public string? JobLevelTitle { get; set; }
        public string? LocationTitle { get; set; }

        // لیست کامل راه‌های ارتباطی
        public List<ContactDetailDto> Contacts { get; set; } = new();

        // ۱. رشته ترکیب‌شده شماره‌ها برای سطر اصلی (با -)
        public string ContactSummary => string.Join(" - ", Contacts.Where(t=>t.Type == ContactTypeEnum.Phone 
        || t.Type == ContactTypeEnum.Mobile
        || t.Type == ContactTypeEnum.OrganizationMobile
        || t.Type == ContactTypeEnum.OfficePhone
        ).Select(c => c.Value));

        // ۲. فلاگ کنترل‌کننده آکاردئون در فرانت‌اند
        public bool HasMultipleContacts => Contacts.Count > 1;
    }
}

File: FrontEnd/src/modules/Contact/models/PhoneBookEmploymentDto.ts
// models/PhoneBookEmploymentDto.ts

export interface PhoneBookEmploymentDto {
  uniqueKey: string; 
  employmentCode: string;
  postCode: string;
  firstName?: string | null;
  lastName?: string | null;
  fullName?: string | null;
  
  organizationUnitsName: string[];
  headOfOrganizationUnitsName: string[];
  jobTitleName: string[];
  jobLevelTitle: string[];
  locationTitle: string[];
  
  
  profilePictureUrl?: string | null;
  gender?: GenderEnum | null;

  contactSummary?: string | null;
  // hasMultipleContacts?: boolean | false;

  contacts?: ContactDetailDto[] | null;
}
export interface ContactDetailDto {
  title: string;
  value: string;
  type?: ContactTypeEnum | null;
  source?: ContactSourceEnum | null;
 

}
export enum GenderEnum {
  Male = 1,
  Female = 2,
  Other = 3
}
export enum ContactTypeEnum {
  Mobile = 1 ,
    Phone = 2,
    OfficePhone = 3,
    OrganizationMobile = 4,
    Email = 5,
 Fax = 6,            // فکس
 Website = 7,        // وب‌سایت
 WhatsApp = 8,       // واتس‌اپ
 Instagram = 9,      // اینستاگرام
 Telegram = 10,       // تلگرام
 LinkedIn = 11,      // لینکدین
 Address = 12, // آدرس پستی یا لوکیشن
 PostalCode = 13, // کد پستی
 Other = 99          // سایر راه ارتباطی
}

export enum ContactSourceEnum {
 Personal= 0,   
      post= 1, 
      employment= 2, 
      location = 3 
 
}
